rfcs/0001-messages.md

```diff
@@ -0,0 +1,146 @@
+# FROST messages
+
+Proposes a message layout to exchange information between participants of a FROST setup.
+
+## Motivation
+
+Currently FROST library is complete for 2 round signatures with a dealer/aggregator setup.
+This proposal is only considering that specific features, additions and upgrades will need to be made when DKG is implemented.
+
+Assuming all participants have a FROST library available we need to define message structures in a way that data can be exchanged between participants. The proposal is a collection of data types so each side can do all the actions needed for a real life situation.
+
+## Definitions
+
+- `dealer`
+- `aggergator`
+- `signer`
+- `nonce`
+- `commitment`
+- 
+
+## Guide-level explanation
+
+We propose a message separated in 2 parts, a header and a payload:
+
+```rust
+struct Message {
+    header: Header,
+    payload: Payload,
+}
+```
+
+`Header` will look as follows:
+
+```rust
+struct Header {
+    msg_type: MsgType,
+    version: MsgVersion,
+    sender: Participant,
+    receiver: Participant, 
+}
+```
+
+While `Payload` will be defined as:
+
+```rust
+enum Payload {
+    DealerBroadcast(MsgDealerBroadcast),
+    Commitments(MsgCommitments),
+    SigningPackage(MsgSigningPackage),
+    SignatureShare(MsgSignatureShare),
+    FinalSignature(MsgFinalSignature),
+}
+```
+
+All the messages and new types will be defined in a new file `src/frost/messages.rs`
+
+## Reference-level explanation
+
+Here we explore in detail the header types and all the message payloads. 
+
+### Header
+
+Fields of the header define new types. Proposed implementation for them is as follows:
+
+```rust
+enum MsgType {
+    DealerBroadcast = 1,
+    Commitments = 2,
+    SigningPackage = 3,
+    SignatureShare = 4,
+    FinalSignature = 5,
+}
+
+enum MsgVersion {
+    FirstVersion = 1,
+    ..
+}
+
+struct Participant(u8);
+```
+
+### Payloads
+
+Each payload defines a new message:
+
+```rust
+// The dealer broadcast initial data, dealer must send
+// this message to each participant involved.
+struct MsgDealerBroadcast {
+    share: frost::SharePackage,
```

Comment:
I'm not sure if serializing and sending the whole `SharePackage` is the correct way since this struct contains other data that would be either repeated or not needed in the message. For example, the receiver ID. Similarly for the other message types. I think we need more abstract structures.